libs/state/selections/src/lib/operators/select.ts

```diff
@@ -234,6 +235,26 @@ export function select<T extends Record<string, unknown>>(
       );
     } else if (isOperateFnArrayGuard(opOrMapFn)) {
       return state$.pipe(stateful(pipeFromArray(opOrMapFn)));
+    } else if (isStringArrayWithoutFunctionAndOptionalTupleGuard(opOrMapFn)) {
+      return state$.pipe(
+        selectSlice<T & object, keyof T>(
+          opOrMapFn[0] as (keyof T)[],
+          opOrMapFn[2] as KeyCompareMap<{ [P in keyof T]: (T & object)[P] }>
+        ),
+        stateful(
+          map((pickSlice: PickSlice<T & Object, keyof T>) => {
+            return opOrMapFn[0].reduce(
+              // @ts-ignore
+              (acc: { [key: keyof T]: T }, key: keyof T) => {
+                // @ts-ignore
+                acc[key] = pickSlice[key as keyof T];
+                return acc;
+              },
+              {}
+            );
+          })
+        )
```

Comment:
question: is this actually needed?

Couldn't it be just:

```ts
return state$.pipe(
        selectSlice(
          opOrMapFn[0],
          opOrMapFn[2]
        )
      );
```
???

At least your test passes that way :)